packages/discord.js/library/use-message.ts

```diff
@@ -0,0 +1,5 @@
+import type { Message } from "discord.js";
+
+export function useMessage(): Message {
+  return {} as Message;
+}
```

Comment:
To make this work, the DJS adapter could render its own context provider with whatever content the user passes:
```tsx
// reacord-discord-js.tsx
export const MessageContext = createContext<Message | undefined>(undefined)

export class ReacordDiscordJs extends Reacord {
	// ...

	protected createInstance(renderer: Renderer): ReacordInstance {
		const { render, ...instance } = super.createInstance(renderer)
		return {
			...instance,
			render(content) {
				return render(
					// renderer.message needs to be public now
					<MessageContext.Provider value={renderer.message}>
						{content}
					</MessageContext.Provider>,
				)
			},
		}
	}

	// ...
}
```
Although 🤔 since this is a separate package, that means `MessageContext` has to become a public export of the library... not great.

I guess we could just mark it as `@private` in JSDoc, but the "real" solution would be to move `ReacordDiscordJs` to this new package, so it can use that context without having to export it publicly

Reply:
Yes I was thinking about a similar solution, this is why I left it empty. I will investigate the problem further. Thank you for an example